pyuvdata/uvflag/tests/test_uvflag.py

```diff
@@ -2795,6 +2795,50 @@ def test_select(input_uvf, uvf_mode):
         )
 
 
+@pytest.mark.parametrize("uvf_mode", ["to_flag", "to_metric"])
+def test_parse_ants_error(uvf_from_uvcal, uvf_mode):
+    uvf = uvf_from_uvcal
+    # used to set the mode depending on which input is given to uvf_mode
+    getattr(uvf, uvf_mode)()
+    err_msg = (
+        "UVFlag objects can only call 'parse_ants' function if type is 'baseline'."
+    )
+    with pytest.raises(ValueError, match=err_msg):
+        uvf.parse_ants("all")
+
+
+@pytest.mark.parametrize(
+    "select_kwargs,err_msg",
+    [
+        (
+            {"ant_str": "all", "antenna_nums": [1, 2, 3]},
+            "Cannot provide ant_str with antenna_nums, bls, or polarizations.",
+        ),
+        (
+            {"ant_str": "all", "bls": [(0, 1), (1, 2)]},
+            "Cannot provide ant_str with antenna_nums, bls, or polarizations.",
+        ),
+        (
+            {"ant_str": "all", "polarizations": [-5, -6, -7]},
+            "Cannot provide ant_str with antenna_nums, bls, or polarizations.",
+        ),
+        (
+            {"ant_str": "auto"},
+            "There is no data matching ant_str=auto in this object.",
+        ),
+    ],
+)
+@pytest.mark.parametrize("uvf_mode", ["to_flag", "to_metric"])
+def test_select_parse_ants_errors(uvf_from_data, uvf_mode, select_kwargs, err_msg):
+    uvf = uvf_from_data
+    # used to set the mode depending on which input is given to uvf_mode
+    getattr(uvf, uvf_mode)()
+    if select_kwargs["ant_str"] == "auto":
```

Comment:
This block looks counterintuitive. Is there something subtle here or is it a typo? We will be able to use ant_str to get autos, right?

Reply:
Sorry if it is not clear enough, this is a pretty contrived example in the test where I am fishing to raise and error in the `select` function. So I call select with `ant_str=auto` but before doing so call it with `ant_str=cross` to ensure the auto ant_str will not have any matching data on the object.

Reply:
Just to explicitly clear it will be possible to use ant_str to get autos in general. Should behave exactly like the one in uvdata.